atom/browser/atom_browser_client.cc

```diff
@@ -140,8 +185,7 @@ void AtomBrowserClient::OverrideSiteInstanceForNavigation(
     return;
   }
 
-  // Restart renderer process for all navigations except "javacript:" scheme.
-  if (url.SchemeIs(url::kJavaScriptScheme))
+  if (!ShouldCreateNewSiteInstance(browser_context, current_instance, url))
```

Comment:
> What do you think about patching libchromiumcontent to notify AtomBrowserClient whenever a new RenderProcessHost is created?

It depends on how large the patch would be since we need to maintain the patch for every Chrome update, if it just inserts a simple hook like other patches, then I'm 👍 with it.